File: Labs/lab2/CalcPrefSum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab2
{
	public class CalcPrefSum
	{
		// Функція для обчислення залишкової кількості монет у стопках
		public static int[] CalculateRemainingCoins(int[] coins, int N)
		{
			int[] sum = new int[N + 1];
			for (int i = N - 1; i >= 0; i--)
			{
				sum[i] = sum[i + 1] + coins[i];  // Сума монет від стопки i до кінця
			}
			return sum;
		}
	}
}

File: Labs/lab13/src/Labs/CalcPrefSum.ts
// CalcPrefSum.ts
export class CalcPrefSum {
    // Function to calculate the remaining coins in stacks
    static calculateRemainingCoins(coins: number[], N: number): number[] {
      const sum = new Array(N + 1).fill(0);
      for (let i = N - 1; i >= 0; i--) {
        sum[i] = sum[i + 1] + coins[i]; // Sum of coins from stack i to the end
      }
      return sum;
    }
  }
  